packages/@tinacms/cli/src/cmds/init/apply.ts

```diff
@@ -0,0 +1,795 @@
+    // importStatements are the import statements that will be added to the tina config
+    // templateCodeString is the string that will be added to the template.{ts,js} file
+    const res = await forestryMigrate({
+      frontMatterFormat: env.frontMatterFormat || config.frontMatterFormat,
+      pathToForestryConfig,
+      usingTypescript: config.typescript,
+    })
+    if (res) {
+      templateCode = res.templateCodeString
+      collections = res.collectionString
+      extraText = res.importStatements
+      isForestryMigration = true
+    }
+  }
+
+  // Report telemetry
+  await reportTelemetry({
+    usingTypescript: config.typescript,
+    hasForestryConfig: env.forestryConfigExists,
+    noTelemetry: noTelemetry,
+  })
+
+  // if no package.json, init
+  if (!env.packageJSONExists) {
+    await createPackageJSON()
+  }
+
+  // if no .gitignore, create one
+  if (!env.gitIgnoreExists) {
+    await createGitignore({ baseDir })
+  } else {
+    let itemsToAdd = []
+    if (!env.gitIgnoreNodeModulesExists) {
+      itemsToAdd.push('node_modules')
+    }
+    if (!env.gitIgnoreEnvExists) {
+      itemsToAdd.push('.env')
+    }
+    if (itemsToAdd.length > 0) {
+      await updateGitIgnore({ baseDir, items: itemsToAdd })
+    }
+  }
+
+  if (isForestryMigration && !env.tinaConfigExists) {
+    await addTemplateFile({
+      generatedFile: env.generatedFiles['templates'],
+      content: templateCode,
+      config,
+    })
+  }
+  const usingDataLayer = config.hosting === 'self-host'
+
+  if (usingDataLayer) {
+    await addDatabaseFile({
+      config,
+      generatedFile: env.generatedFiles['database'],
+    })
+    await addGqlApiHandler({
+      config,
+      generatedFile: env.generatedFiles['gql-api-handler'],
+    })
+  }
+
+  if (config.nextAuthProvider) {
+    await addAuthFile({
+      config,
+      generatedFile: env.generatedFiles['auth'],
+    })
+
+    await addNextAuthApiHandler({
+      config,
+      generatedFile: env.generatedFiles['next-auth-api-handler'],
+      content: NextTemplates['next-auth-api-handler'](),
+    })
+
+    if (config.nextAuthProvider === 'vercel-kv-credentials-provider') {
+      await addVercelKVCredentialsProviderFiles({
+        generatedSignin:
+          env.generatedFiles['vercel-kv-credentials-provider-signin'],
+        generatedRegister:
+          env.generatedFiles['vercel-kv-credentials-provider-register'],
+        generatedTailwindCSS:
+          env.generatedFiles['vercel-kv-credentials-provider-tailwindcss'],
+        generatedRegisterApiHandler:
+          env.generatedFiles[
+            'vercel-kv-credentials-provider-register-api-handler'
+          ],
+        generatedTinaSVG: env.generatedFiles['tina.svg'],
+        config,
+      })
+    }
+  }
+
+  if (!env.forestryConfigExists && !env.tinaConfigExists) {
+    // add /content/posts/hello-world.md
+    await addContentFile({ config, env })
+  }
+
+  if (config.framework.reactive && !env.tinaConfigExists) {
```

Comment:
@logan-anderson I am working on tests for init and running into issues with this change: https://github.com/tinacms/tinacms/commit/f9bf419e1bca49431569e739fa918a29308f332c

The change adds skipping the generation of files if the folder tina is already present. When you run the backend init and self-host, in the addReactiveFile function, this results in the demo blog file not having the databaseClient and the `setup:users` script won't ever get added to the package.json because the vercel-kv-credentials-provider option only gets set during `init backend`.

If we make the blog post file a GeneratedFile and use writeGeneratedFile, then it will prompt the user whether to overwrite it, and during backend init the correct template will get generated (assuming the user says yes to overwrite). Additionally the extendNextScripts() function can be safely re-run on the same package.json. So I think the `!env.tinaConfigExists` guard on the call to addReactiveFile should probably be removed and the demo blog post should become a GeneratedFile.